apollo-client/src/main/java/com/ctrip/framework/apollo/util/OrderedProperties.java

```diff
@@ -79,4 +73,25 @@ private void addPropertyName(Object key) {
       propertyNames.add((String) key);
     }
   }
+
+  @Override
+  public synchronized void putAll(Map<?, ?> t) {
```

Comment:
I think `putAll` is unnecessary?

Reply:
Sorry for running tests on JDK10, and add this method to fix tests failure. Just notice that Properties are different in JDK1.8/1.7 and JDK 10. The `putAll` method is inherited from `Properites` class in JDK 10 while it is from `Hashtable` in JDK1.8/1.7. 
Remove it in lastest commit.

Reply:
This method is required in JDK10, due to code in `RemoteConfigRepository` L163 invoking putAll.

``` JAVA
  private Properties transformApolloConfigToProperties(ApolloConfig apolloConfig) {
    Properties result =  PropertiesFactory.getPropertiesObject();     
    result.putAll(apolloConfig.getConfigurations());
    return result;
  }
```

Reply:
You are right, the implementation of `Properties` in JDK 10 is totally different from that in JDK 7/8...So `putAll` is required.

And for this feature, it is important we test it against JDK 7, 8, 11.

Reply:
I think apollo is not ready for JDK 11, because UT will fail on JDK 11. 
So should we add putAll now, or add it until all are prepared for JDK11?

Reply:
Actually apollo is ok with JDK 11, which we have tested functionally.
(There is one issue brought by #2828, which is fixed in #2888)

However, the UT does fail with JDK 11 and I will try to fix it later.

So in a word, we need to test the function againt JDK 7, 8 and 11.